src/hooks/useUserReputation.ts

```diff
@@ -0,0 +1,53 @@
+import { gql, useQuery } from '@apollo/client';
+import { Id } from '@colony/colony-js';
+
+import { Address } from '~types';
+import { ADDRESS_ZERO } from '~constants';
+import { getUserReputation } from '~gql';
+
+interface UseUserReputationHook {
+  userReputation?: string;
+  totalReputation?: string;
+}
+
+const useUserReputation = (
+  colonyAddress?: Address,
+  walletAddress?: Address,
+  domainId = Id.RootDomain,
+  rootHash?: string,
+): UseUserReputationHook => {
+  const { data: userReputationData } = useQuery(gql(getUserReputation), {
+    variables: {
+      input: {
+        colonyAddress: colonyAddress ?? '',
+        walletAddress: walletAddress ?? '',
+        domainId,
+        rootHash,
+      },
+    },
+    fetchPolicy: 'cache-and-network',
```

Comment:
I'm not sure `cache-and-network` works here, as I think it should be `network-only` to get the freshest reputation always, but I don't have enough arguments for one of the other.

So for the time being leave it as it is, and if we notice it doesn't work as we expect it, then we'll address it.

Reply:
I believe `cache-and-network` will eventually get the freshest reputation with the only exception being it will show the cached value (which I feel like would be correct most of the time) while it performs the network request. I didn't actually specify this myself but rather copied it from the dapp